language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java

```diff
@@ -536,4 +537,25 @@ public static int toRank(BallerinaCompletionContext context, LSCompletionItem co
 
         return rank;
     }
+
+    /**
+     * Checks whether the symbol completion item is within the range of given node and cursor.
+     *
+     * @param context       Completion Context
+     * @param lsCItem       LS Completion Item
+     * @param startNode     Starting Node
+     * @return  {@link Boolean}
+     */
+    public static boolean isSymbolCItemWithinNodeAndCursor(BallerinaCompletionContext context, 
+                                                               LSCompletionItem lsCItem, Node startNode) {
+        if (lsCItem.getType() != LSCompletionItem.CompletionItemType.SYMBOL 
+                || ((SymbolCompletionItem) lsCItem).getSymbol().isEmpty()) {
+                return false;
+        }
+
+        Optional<Location> symbolLocation = ((SymbolCompletionItem) lsCItem).getSymbol().get().getLocation();
```

Comment:
Don't we need a `isEmpty()` before calling `get()`?

Reply:
It is already checked

Reply:
Ah, my bad. Its already checked